File: src/main/java/com/rs/java/game/area/banks/ArdougneNorthBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.Shape;
import com.rs.java.game.area.shapes.Polygon;

public final class ArdougneNorthBank extends Area {

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Ardougne North Bank";
	}

	@Override
	public Shape[] shapes() {
		return new Shape[] { new Polygon(new WorldTile[] {
				new WorldTile(2612, 3330, 0),
				new WorldTile(2612, 3335, 0),
				new WorldTile(2621, 3335, 0),
				new WorldTile(2621, 3330, 0),
				new WorldTile(2612, 3330, 0),
		}) };
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}

}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/ArdougneNorthBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class ArdougneNorthBank : Area(
    Polygon(
        arrayOf(
            WorldTile(2612, 3330, 0),
            WorldTile(2612, 3335, 0),
            WorldTile(2621, 3335, 0),
            WorldTile(2621, 3330, 0),
            WorldTile(2612, 3330, 0) // closed polygon
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Ardougne North Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
